consensus/state.go

```diff
@@ -1104,20 +1109,27 @@ func (cs *State) isProposer(address []byte) bool {
 
 func (cs *State) defaultDecideProposal(height int64, round int32) {
 	var (
-		block      *types.Block
-		blockParts *types.PartSet
+		block               *types.Block
+		propBlockID         types.BlockID
+		consensusBlockParts *types.PartSet
 	)
 
 	// Decide on block
 	if cs.ValidBlock != nil {
 		// If there is valid block, choose that.
-		block, blockParts = cs.ValidBlock, cs.ValidBlockParts
+		block = cs.ValidBlock
+		propBlockID = types.BlockID{
+			Hash:          block.Hash(),
+			PartSetHeader: cs.ValidBlockParts.Header(),
+		}
+		consensusBlockParts = cs.blockExec.ConsensusPartSetFromBlock(block)
```

Comment:
I don't think this is necessary... if we have a valid block, then consensus for the block has already surpassed the set proposal stage. However, I'm not sure how that plays with the rest of the consensus just yet. The consensus block parts are quite tiny, and in the ase of CList mempool, consensus block parts == ProposalBlockParts (full block parts)...